src/templates/modules/LM_PC_Template_v1.sol

```diff
@@ -0,0 +1,190 @@
+            || super.supportsInterface(interfaceId_);
+    }
+
+    //--------------------------------------------------------------------------
+    // Constants
+
+    uint internal constant _maxDepositAmount = 100 ether;
+
+    //--------------------------------------------------------------------------
+    // State
+
+    /// @dev The role that allows processing deposits
+    bytes32 public constant DEPOSIT_ADMIN_ROLE = "DEPOSIT_ADMIN";
+
+    /// @notice    Mapping of user addresses to their deposited token amounts.
+    mapping(address user => uint amount) internal _depositedAmounts;
+
+    /// @notice    Payment token.
+    IERC20 internal _paymentToken;
+
+    /// @notice    Storage gap for future upgrades.
+    uint[50] private __gap;
+
+    // =========================================================================
+    // Modifiers
+
+    modifier onlyValidDepositAmount(uint amount_) {
+        _ensureValidDepositAmount(amount_);
+        _;
+    }
+
+    // =========================================================================
+    // Constructor & Init
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        __Module_init(orchestrator_, metadata_);
+
+        // Decode module specific init data through use of configData bytes.
+        // This value is an example value used to showcase the setters/getters
+        // and internal functions/state formatting style.
+        (address paymentToken) = abi.decode(configData_, (address));
+
+        // Set init state.
+        _paymentToken = IERC20(paymentToken);
+    }
+
+    // =========================================================================
+    // Public - Mutating
+
+    /// @inheritdoc ILM_PC_Template_v1
+    function deposit(uint amount_)
+        external
+        virtual
+        onlyValidDepositAmount(amount_)
+    {
+        // Update state.
+        _depositedAmounts[_msgSender()] += amount_;
+
+        // Transfer tokens.
+        _paymentToken.safeTransferFrom(_msgSender(), address(this), amount_);
+
+        // Emit event.
+        emit Deposited(_msgSender(), amount_);
+    }
+
+    /// @inheritdoc ILM_PC_Template_v1
+    function processDeposit(address user_)
+        external
+        onlyModuleRole(DEPOSIT_ADMIN_ROLE)
+    {
+        uint amount = _depositedAmounts[user_];
+
+        // Clear the deposit amount before processing
+        _depositedAmounts[user_] = 0;
+
+        // Create and add payment order
+        PaymentOrder memory order = PaymentOrder({
+            recipient: user_,
+            paymentToken: address(_paymentToken),
+            amount: amount,
+            originChainId: block.chainid,
+            targetChainId: block.chainid,
+            flags: 0,
+            data: new bytes32[](0)
+        });
+
+        _addPaymentOrder(order);
+
+        // Process the payment
```

Comment:
Even if the comment is a bit pointless (it doesn't add any information, but i see the point of wanting to show that we add comments where it makes sense).
```suggestion
        // Process the payment.
```